javaslang/src/main/java/javaslang/concurrent/Future.java

```diff
@@ -309,10 +309,45 @@
      * @throws NullPointerException if one of executorService of computation is null.
      */
     static <T> Future<T> of(ExecutorService executorService, CheckedSupplier<? extends T> computation) {
+        return Future.of(executorService, computation, 0);
+    }
+
+    /**
+     * Starts an asynchronous computation after waiting a duration, backed by the {@link #DEFAULT_EXECUTOR_SERVICE}.
+     * The thread supplied by the {@code DEFAULT_EXECUTOR_SERVICE} will be blocked at least {@code duration} milliseconds.
+     * If the {@code duration} &lt;= 0, the {@code computation} will be executed immediately.
+     *
+     * @param computation A computation.
+     * @param duration    The time in milliseconds to wait before executing the given {@code computation}.
+     * @param <T>         Type of the computation result.
+     * @return A new Future instance.
+     * @throws NullPointerException if computation is null.
+     */
+    static <T> Future<T> of(CheckedSupplier<? extends T> computation, long duration) {
```

Comment:
Yes, right! (I didn't like the name 'timeout' of JavaScript because it intends to cancel the computation after a specific amount of time.)

Delay is much better than duration.